src/components/pdf-viewer/layers/util/generateHighlights/__tests__/generateHighlights.test.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import path from 'path'
+
+import { generateHighlights, flattenDOMNodes } from '../index'
+
+describe('PDF viewer utilities', () => {
+    it('checks if all marks match the query', () => {
+        for (let index = 1; index < 3; index++) {
+            document.body.innerHTML = '' // reset document.body between files / use cases
+            const fileContent = require('fs').readFileSync(path.join(__dirname, `./${index}.html`))
+            document.body.innerHTML = fileContent
+
+            const marks = generateHighlights([{ pageNum: 0, index: 0 }], document.body.children[0], 'metod', { pageNumber: 0 }, 0)
```

Comment:
Can we simplify this and remove all the `[{ pageNum: 0, index: 0 }]` and `{ pageNumber: 0 }, 0` from this function interface? A lot of zeros there...

I think it might be a good idea to separate adding the marks to a page (implemented here) from setting and changing the selected orange mark that you scroll with next/prev.

That way, this test is cleaner, and the next/prev button doesn't need to re-run the matcher algorithm for each button press, just find and switch the orange mark to next/prev position. Which can be tested separately with some simple data.

Reply:
Very good idea, thanks! Done, and highlight navigation is faster now.